cyberdrop_dl/scraper/crawlers/coomer_crawler.py

```diff
@@ -250,26 +227,108 @@ async def get_user_info(self, scrape_item: ScrapeItem) -> dict:
+            profile_json, resp = await self.client.get_json(
+                self.domain, profile_api_url, origin=scrape_item, cache_disabled=True
+            )
+            properties = profile_json.get("props", {})
+            cached_response = await self.manager.cache_manager.request_cache.get_response(str(profile_api_url))
+            cached_properties = {} if not cached_response else (await cached_response.json()).get("props", {})
+
+            # Shift cache offsets if necessary
+            await self.shift_offsets(profile_api_url, properties, cached_properties, resp)
 
+        limit = properties.get("limit", 50)
         user_str = properties.get("name", user)
         if post:
             offset, maximum_offset = None, None
         else:
             offset = int(scrape_item.url.query.get("o", 0))
-            maximum_offset = int(properties.get("count", 0))
+            maximum_offset = maximum_offset = (int(properties.get("count", 0)) // limit) * limit
 
-        returnValues = {
+        return {
             "service": service,
             "user": user,
             "post": post,
             "user_str": user_str,
             "offset": offset,
             "maximum_offset": maximum_offset,
+            "limit": limit,
         }
 
-        return returnValues
+    async def shift_offsets(self, api_url: URL, new_properties: dict, cached_properties: dict, response) -> None:
+        """
+        Adjust cached responses for shifted offsets based on the difference in the number of posts.
+
+        Args:
+            api_url (URL): The legacy API URL.
+            new_properties (dict): Properties from the current response.
+            cached_properties (dict): Properties from the cached response.
+            response: The latest HTTP response to save to the cache.
+
+        Returns:
+            int: The updated maximum offset.
+        """
+        user_str = new_properties.get("name", "Unknown")
+        new_count = int(new_properties.get("count", 0))
+        cached_count = int(cached_properties.get("count", 0))
+        if cached_count == 0:
+            return
+        post_limit = int(new_properties.get("limit", 50))
+        shift = new_count - cached_count
+
+        if shift > 0:
+            log(f"{shift} new posts detected for {user_str} (Coomer). Adjusting cache...", 20)
+
+            cached_posts = []
+            offset = 0
+            while True:
+                paginated_api_url = api_url.with_query({"o": offset})
+                cache_key = self.manager.cache_manager.request_cache.create_key("GET", paginated_api_url)
+                cached_response = await self.manager.cache_manager.request_cache.get_response(cache_key)
+
+                if not cached_response:
+                    break  # Stop if no cached response exists for the current offset
+
+                cached_json = await cached_response.json()
+                cached_posts.extend(cached_json)
+                offset += post_limit
```

Comment:
Instead of using a `while` and `break`, use a `for` to loop trough all the possible URLs based of `new_count`.

The user may have stopped CDL in the middle of a shift operation and that will leave the database with missing response entries. The next time it tries to shift responses, it will end prematurely. Any other possible existing response after it will need to be invalidated as well cause we won't know how to shift its posts.

```suggestion
            invalidate = False
            max_offset = (new_count // post_limit ) * post_limit
            for offset in range(0, max_offset + 1, post_limit):
                paginated_api_url = api_url.with_query({"o": offset})
                cache_key = self.manager.cache_manager.request_cache.create_key("GET", paginated_api_url)
                cached_response = await self.manager.cache_manager.request_cache.get_response(cache_key)

                if not cached_response:
                    invalidate = True
                    continue
                    
                if invalidate:
                    await self.manager.cache_manager.request_cache.delete_url(cache_key)
                    log(f"Invalidated cached page: {paginated_api_url}", 20)
                    continue

                cached_json = await cached_response.json()
                cached_posts.extend(cached_json)
```